src/routers/controllers/confirmCompanyController.ts

```diff
@@ -17,10 +17,9 @@ export const confirmCompanyControllerGet = async (req: Request, res: Response):
 export const confirmCompanyControllerPost = async (req: Request, res: Response): Promise<void> => {
     const session: Session = req.session as Session;
     const company = session.data.extra_data.companyProfile;
-    const userEmailAddress = getLoggedInUserEmail(req.session);
     const companyNotActive = company.companyStatus.toLocaleLowerCase() !== constants.COMPANY_STATUS_ACTIVE;
 
-    const isAlreadyAssociated = await isCompanyAssociatedWithUser(company.companyNumber, userEmailAddress);
+    const isAlreadyAssociated = await isCompanyAssociatedWithUser(req, company.companyNumber);
```

Comment:
is already associated can be done using the new endpoint which takes email as filter.

/associations/{company_number}?yser_email=<email>

Reply:
Do you mean /associations/companies/{company_number}?user_email=

Reply:
I've changed the isCompanyAssociatedWithUser function so it uses /associations/companies/{company_number}?user_email= endpoint